gobblin-runtime/src/main/java/gobblin/runtime/TaskState.java

```diff
@@ -83,21 +83,20 @@
   private long duration;
 
   // Needed for serialization/deserialization
-  public TaskState() {
-  }
+  public TaskState() {}
 
   public TaskState(WorkUnitState workUnitState) {
     // Since getWorkunit() returns an immutable WorkUnit object,
     // the WorkUnit object in this object is also immutable.
-    super(workUnitState.getWorkunit());
+    super(workUnitState.getWorkunit(), workUnitState.getJobState());
     addAll(workUnitState);
```

Comment:
I agree, I think `WorkUnitState` and `SourceState` are useless, they are the same as `TaskState` and `JobState`, respectively. Also, `Extract` should also be deprecated as it is conceptually identical as `Dataset`. These changes will be part of another PR.